services/configuration/src/errors.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::blockchain::{ExecutionError, StoredConfiguration};
+use exonum::crypto::Hash;
+use exonum::encoding::serialize::json::reexport::Error as JsonError;
+use exonum::helpers::Height;
+
+use transactions::Propose;
+
+// Implements conversion between two field-less enums.
+macro_rules! convert_codes {
+    ($from:ident => $to:ident { $($variant:ident,)+ }) => {
+        #[doc(hidden)]
+        impl From<$from> for $to {
+            fn from(value: $from) -> $to {
+                match value {
+                    $($from::$variant => $to::$variant,)+
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+#[repr(u8)]
+enum CommonErrorCode {
```

Comment:
I'm not sure how to deal with common error codes. Any suggestions how to minimize copypaste would be appreciated.